CP panics if an HTTPRoute references a Gateway with a nonexistent GatewayClass

### What happened?

I tried to follow the [Gateway API guide](https://gateway-api.sigs.k8s.io/guides/) (maybe with a little bit different order, as I first installed the [Gateway API resources](https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.6.2/experimental-install.yaml), then added Gateway + HTTPRoute, and then installed Kuma with gatewayapi flag). When doing so I ended with Kuma CP panicking with nil pointer exception.

## Details

Kuma Version: 2.2.0

### Steps to reproduce

1. Create an empty k3d cluster
```sh
pushd ~/kuma
make k3d/start KIND_CLUSTER_NAME=kuma-2
```

2. Install [Gateway API resources](https://gateway-api.sigs.k8s.io/guides/#install-experimental-channel)
```sh
kubectl apply -f https://github.com/kubernetes-sigs/gateway-api/releases/download/v0.6.2/experimental-install.yaml
```

3. Import kuma-universal image which will be used for deploying simple http server  using `test-server` binary
```sh
k3d image import kumahq/kuma-universal:0.0.0-preview.vlocal-build --cluster kuma-2
```

4. Deploy simple http server
```sh
echo '
---
apiVersion: v1
kind: Namespace
metadata:
  name: example
  labels:
    kuma.io/sidecar-injection: enabled
---
apiVersion: v1
kind: Service
metadata:
  name: example-server
  namespace: example
spec:
  ports:
  - port: 80
    name: http
    appProtocol: http
  selector:
    service: example-server
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: example-server
  namespace: example
  labels:
    service: example-server
spec:
  selector:
    matchLabels:
      service: example-server
  template:
    metadata:
      labels:
        service: example-server
    spec:
      terminationGracePeriodSeconds: 0
      containers:
      - name: test-server
        image: "kumahq/kuma-universal:0.0.0-preview.vlocal-build"
        imagePullPolicy: IfNotPresent
        command:
        - "test-server"
        args:
        - "echo"
        - "--port"
        - "80"
        ports:
        - containerPort: 80
' | kubectl apply -f-
```

5. Deploy [Gateway resources](https://gateway-api.sigs.k8s.io/guides/simple-gateway/#deploying-a-simple-gateway)
```sh
echo 'apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: prod-web
  namespace: example
spec:
  gatewayClassName: acme-lb
  listeners:
  - protocol: HTTP
    port: 80
    name: prod-web-gw
    allowedRoutes:
      namespaces:
        from: Same' | kubectl apply -f-

echo 'apiVersion: gateway.networking.k8s.io/v1beta1
kind: HTTPRoute
metadata:
  name: foo
  namespace: example
spec:
  parentRefs:
  - name: prod-web
  rules:
  - backendRefs:
    - name: example-server
      port: 80' | kubectl apply -f-
```

6. Install Kuma with `--experimental-gatewayapi` flag
```sh
kumactl install control-plane --experimental-gatewayapi | kubectl apply -f-
```

### Kuma CP logs

```sh
2023-05-10T04:35:32.346Z    INFO    kube-manager    Starting workers    {"controller": "mesh", "controllerGroup": "kuma.io", "controllerKind": "Mesh", "worker count": 1}
2023-05-10T04:35:32.346Z    INFO    kube-manager    Starting workers    {"controller": "mesh", "controllerGroup": "kuma.io", "controllerKind": "Mesh", "worker count": 1}
2023-05-10T04:35:32.346Z    INFO    kube-manager    Starting workers    {"controller": "namespace", "controllerGroup": "", "controllerKind": "Namespace", "worker count": 1}
2023-05-10T04:35:32.346Z    INFO    kube-manager    Starting workers    {"controller": "gatewayclass", "controllerGroup": "gateway.networking.k8s.io", "controllerKind": "GatewayClass", "worker count": 1}
2023-05-10T04:35:32.364Z    ERROR    kube-manager    Reconciler error    {"controller": "gatewayclass", "controllerGroup": "gateway.networking.k8s.io", "controllerKind": "GatewayClass", "GatewayClass": {"name":"kuma"}, "namespace": "", "name": "kuma", "reconcileID": "d8c34b64-886f-4629-b7ab-923e8e07fe54", "error": "Ope
ration cannot be fulfilled on gatewayclasses.gateway.networking.k8s.io \"kuma\": the object has been modified; please apply your changes to the latest version and try again"}
2023-05-10T04:35:32.367Z    INFO    defaults    trying to create default Mesh
2023-05-10T04:35:32.367Z    INFO    kube-manager    Starting workers    {"controller": "meshgatewayinstance", "controllerGroup": "kuma.io", "controllerKind": "MeshGatewayInstance", "worker count": 1}
2023-05-10T04:35:32.368Z    INFO    kube-manager    Starting workers    {"controller": "httproute", "controllerGroup": "gateway.networking.k8s.io", "controllerKind": "HTTPRoute", "worker count": 1}
2023-05-10T04:35:32.368Z    INFO    kube-manager    Starting workers    {"controller": "gateway", "controllerGroup": "gateway.networking.k8s.io", "controllerKind": "Gateway", "worker count": 1}
2023-05-10T04:35:32.368Z    INFO    kube-manager    Observed a panic in reconciler: runtime error: invalid memory address or nil pointer dereference    {"controller": "httproute", "controllerGroup": "gateway.networking.k8s.io", "controllerKind": "HTTPRoute", "HTTPRoute": {"name":"foo","namespace":"example"}, "namespace
": "example", "name": "foo", "reconcileID": "9873f1d2-ac7f-4d43-8a96-1b655f7f60bf"}
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x108 pc=0x24a9dca]

goroutine 1529 [running]:
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile.func1()
    sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:119 +0x1fa
panic({0x28296e0, 0x4e91ea0})
    runtime/panic.go:884 +0x213
github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment.getParentRefGateway({0x336c660, 0xc001171920}, {0x337bd10?, 0xc000c123c0?}, {0xc000136440?, 0x30?}, {0xc00130d560, 0xc00130d570, 0x0, {0xc000df60e8, ...}, ...})
    github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment/attachment.go:141 +0x28a
github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment.EvaluateParentRefAttachment({0x336c660?, 0xc001171920?}, {0x337bd10?, 0xc000c123c0?}, {0x0, 0x0, 0x0}, 0xc001ecac60, {0xc00130d560, 0xc00130d570, ...})
    github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment/attachment.go:188 +0xac
github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi.(*HTTPRouteReconciler).gapiToKumaRoutes(0xc0008a27e0, {0x336c660, 0xc001171920}, {0x2d43e83, 0x7}, 0xc001218d80)
    github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go:117 +0x2e5
github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi.(*HTTPRouteReconciler).Reconcile(0xc0008a27e0, {0x336c660, 0xc001171920}, {{{0xc000df60d3, 0x7}, {0xc000df60d0, 0x3}}})
    github.com/kumahq/kuma/pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go:64 +0x385
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile(0x336c660?, {0x336c660?, 0xc001171920?}, {{{0xc000df60d3?, 0x2721e40?}, {0xc000df60d0?, 0x40fb3d?}}})
    sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122 +0xc8
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc00033ab40, {0x336c5b8, 0xc000977e50}, {0x29595a0?, 0xc0016ea1a0?})
    sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323 +0x377
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc00033ab40, {0x336c5b8, 0xc000977e50})
    sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274 +0x1d9
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2()
    sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235 +0x85
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
    sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:231 +0x587
```

--- a/pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment/attachment.go
+++ b/pkg/plugins/runtime/k8s/controllers/gatewayapi/attachment/attachment.go
@@ -138,7 +138,7 @@ func getParentRefGateway(
 		return nil, err
 	}
 
-	if class.Spec.ControllerName != common.ControllerName {
+	if class == nil || class.Spec.ControllerName != common.ControllerName {
 		return nil, nil
 	}
 

